src/develop/develop.c

```diff
@@ -1420,7 +1420,12 @@ static gboolean _dev_auto_apply_presets(dt_develop_t *dev)
   const gboolean auto_apply_cat = has_matrix && is_modern_chroma;
   const gboolean auto_apply_sharpen = dt_conf_get_bool("plugins/darkroom/sharpen/auto_apply");
 
-  if(auto_apply_filmic || auto_apply_sharpen || auto_apply_cat)
+  // Check if the history has been inited
+  GList *history = dt_history_get_items(imgid, TRUE);
+  const gboolean empty_history = (history == NULL);
+  g_free(history);
```

Comment:
This should be:
`g_list_free_full(history, dt_history_item_free);`

Reply:
But I'm not sure to see the point, does that really fix something ?

You are already checking that the module does not exists on the db before adding it:

`!dt_history_check_module_exists(imgid, module->op)`

So the fact that history is empty or not will not change the behavior I fear.

Reply:
`dt_history_check_module_exists(imgid, module->op)` does a module-wise check and lets us enable the module if not already present. But that may lead to new modules being added on old edits, where they don't belong. What we do here is checking if it's an old edit, and discard the auto-application of everything if there is a previous history at all.

Reply:
So we have another bug somewhere as this case is covered already with the flag DT_IMAGE_AUTO_PRESETS_APPLIED.

See in routine:

`if(!(image->flags & DT_IMAGE_AUTO_PRESETS_APPLIED)) run = TRUE;`

And at the end when this routine has been run:

`  image->flags |= DT_IMAGE_AUTO_PRESETS_APPLIED | DT_IMAGE_NO_LEGACY_PRESETS;`

So the `_dev_auto_apply_presets` cannot be executed twice on the same image.

Reply:
@aurelienpierre : and as noted in #5596  it has been fixed in #6304.

This confirms that this PR is no-op and indeed I cannot reproduce the issue in #5596 with current master.

I think this can be closed.

Reply:
But `DT_IMAGE_AUTO_PRESETS_APPLIED` is set only after the auto-presets has been triggered. We need something to prevent any auto-preset at all if there is an history. I initially wanted to set `DT_IMAGE_AUTO_PRESETS_APPLIED` when opening a new image that has history, but I couldn't find the code.

Reply:
Do you reproduce #5596 with current master on your side ?